consultation_analyser/support_console/views/consultations.py

```diff
@@ -247,40 +306,33 @@ def import_consultation_themes(request: HttpRequest) -> HttpResponse:
                 request, "support_console/consultations/import_themes.html", context=context
             )
 
-        question_part_subfolders = get_all_question_part_subfolders(
-            folder_name=path_to_themes, bucket_name=bucket_name
-        )
+        try:
+            if question_choice == "all":
+                question_part_subfolders = get_all_question_part_subfolders(
+                    folder_name=path_to_themes, bucket_name=bucket_name
+                )
 
-        for folder in question_part_subfolders:
-            question_number = int(folder.split("/")[-2].split("question_part_")[-1])
-            question_part = models.QuestionPart.objects.get(
-                question__consultation=consultation,
-                question__number=question_number,
-                type=models.QuestionPart.QuestionType.FREE_TEXT,
-            )
+                for folder in question_part_subfolders:
+                    question_number = int(folder.split("/")[-2].split("question_part_")[-1])
+                    import_question_part_themefinder_outputs(
+                        consultation, question_number, bucket_name, folder, batch_size
+                    )
 
-            framework = import_themes_from_json_and_get_framework(
-                question_part=question_part,
-                bucket_name=bucket_name,
-                question_part_folder_key=folder,
-            )
-            import_all_theme_mappings_from_jsonl(
-                question_part=question_part,
-                framework=framework,
-                bucket_name=bucket_name,
-                question_part_folder_key=folder,
-                batch_size=batch_size,
-            )
-            import_all_sentiment_mappings_from_jsonl(
-                question_part=question_part,
-                bucket_name=bucket_name,
-                question_part_folder_key=folder,
-                batch_size=batch_size,
-            )
+            else:
```

Comment:
Fine for now, but this naming won't work when we have multiple question parts per question.

I'm not sure how we'll do it for folder naming, something like `question_1_1` and `question_1_2`.

Maybe the input field should then be the name of the folder?

I'll make a ticket for us to think about the best way to do this - and make it more robust.

Reply:
I also think there should be a future version of this where the form has a JS-enabled component that retrieves the available questions from S3 and gives people the choice of which files to select.